src/ReportCommand.php

```diff
@@ -86,30 +117,35 @@ private function groupDependenciesByLicense(array $dependencies)
         return $grouped;
     }
 
-    private function lookUpLicenses(array $licenses, OutputInterface $output, $useCache = true)
+    private function lookUpLicenses(array $licenses, InputInterface $input, OutputInterface $output, $useCache = true): array
     {
         if (! $useCache) {
             $this->licenseLookup->setCache(new NullAdapter);
         }
 
         $lookedUp = [];
         foreach ($licenses as $license) {
-            $output->writeln("Looking up $license ...");
-            $lookedUp[$license] = $this->licenseLookup->lookUp($license);
+            if ($this->isValidLicenseByFilter($license, $input)) {
+                $output->writeln("Looking up $license ...");
+                $lookedUp[$license] = $this->licenseLookup->lookUp($license);
+            }
```

Comment:
I changed it according to your suggestion . I tried to avoid the foreach loops before. I also tried to avoid the continue as you mentioned it. But for your suggestion i don't know how to prevent it .

Reply:
Thanks for updating your solution. I just reviewed it and I'll merge it.

For completion: the solution I had in mind was something like this
```php
private function filterDependenciesByLicenses(array $dependencies, array $relevantLicenses): array
{
    if (empty($relevantLicenses))
        return $dependencies;

    $normalisedLicenses = array_map('strtolower', $relevantLicenses);

    return array_filter($dependencies, function ($dependency) use ($normalisedLicenses) {
        $dependencyLicenses = array_map('strtolower', $dependency->getLicenses());
        return count(array_intersect($dependencyLicenses, $normalisedLicenses)) > 0;
    });
}